client/common/cmdline.c

```diff
@@ -1315,6 +1316,45 @@ static void activate_smartcard_logon_rdp(rdpSettings* settings)
 	freerdp_set_param_bool(settings, FreeRDP_PasswordIsSmartcardPin, TRUE);
 }
 
+/**
+ * parses a string value with the format <v1>x<v2>
+ * @param input: input string
+ * @param v1: pointer to output v1
+ * @param v2: pointer to output v2
+ * @return if the parsing was successful
+ */
+static BOOL parseSizeValue(const char *input, unsigned long *v1, unsigned long *v2)
+{
+	const char *xcharpos;
+	char *endPtr;
+	unsigned long v;
+
+	errno = 0;
+	v = strtoul(input, &endPtr, 10);
+
+	if ((v == 0 || v == ULONG_MAX) && (errno != 0))
+		return FALSE;
+	if (v1)
+		*v1 = v;
+
+	xcharpos = strchr(input, 'x');
+	if (!xcharpos || xcharpos != endPtr)
+		return FALSE;
+
+	errno = 0;
+	v = strtoul(xcharpos + 1, &endPtr, 10);
```

Comment:
Might be a good idea to split the `strtoul` off to a function too (can unify all the uses in command line parser this way)

Reply:
Would suggest something like `BOOL parse_value(const char* input, unsigned long* result, unsigned long max)`